nix/sources.json

```diff
@@ -82,6 +82,18 @@
         "url": "https://nodejs.org/download/release/v10.9.0/node-v10.9.0-headers.tar.gz",
         "url_template": "https://nodejs.org/download/release/<rev>/node-<rev>-headers.tar.gz"
     },
+    "pre-commit-hooks.nix": {
```

Comment:
You need `"builtin": false` so that it uses the nixpkgs fetcher not the builtin one. (This is a bizarre story: the builtin one gets blocked by `allowed-uris`, but the nixpkgs one can get through it because it's a fixed-output derivation...)